solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,511 @@
+  description = "The security group IDs to add to the Virtual server instance deployment secondary interfaces (5 maximum). Use the same value for interface_name as for name in secondary_subnets to avoid applying the default VPC security group on the secondary network interface. [Learn more](https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone-vsi/tree/main/solutions/fully-configurable/DA_inputs.md#options-with-secondary-security-groups)."
+  type = list(
+    object({
+      security_group_id = string
+      interface_name    = string
+    })
+  )
+  default = []
+}
+
+variable "secondary_floating_ips" {
+  description = "List of secondary interfaces to add floating ips"
+  type        = list(string)
+  default     = []
+}
+
+variable "secondary_allow_ip_spoofing" {
+  description = "Allow IP spoofing on additional network interfaces"
+  type        = bool
+  default     = false
+}
+
+##############################################################################
+
+##############################################################################
+# Snapshot Restore Variables
+##############################################################################
+
+variable "boot_volume_snapshot_id" {
+  description = "The snapshot id of the volume to be used for creating boot volume attachment (if specified, the `image_id` parameter will not be used)"
+  type        = string
+  default     = null
+}
+
+variable "snapshot_consistency_group_id" {
+  description = "The snapshot consistency group Id. If supplied, the group will be queried for snapshots that are matched with both boot volume and attached (attached are matched based on name suffix). You can override specific snapshot Ids by setting the appropriate input variables as well."
+  type        = string
+  default     = null
+}
+
+##############################################################################
+# Dedicated Host Variables
+##############################################################################
+
+variable "dedicated_host_id" {
+  type        = string
+  default     = null
+  description = "The ID of the dedicated host for hosting the Virtual server instance's."
+}
+
+##############################################################################
+
+##############################################################################
+## Secrets Manager Service Credentials
+##############################################################################
+
+variable "existing_secrets_manager_instance_crn" {
+  type        = string
+  default     = null
+  description = "The CRN of existing secrets manager to use to store the SSH private key which was auto generated when `auto_generate_ssh_key` was set to true."
+  validation {
+    condition     = var.auto_generate_ssh_key ? var.existing_secrets_manager_instance_crn != null ? true : false : true
+    error_message = "`existing_secrets_manager_instance_crn` is a required value when `auto_generate_ssh_key` is set to true."
```

Comment:
this should not be enforced. Allow the SSH key to be created even if user is not storing the SM